src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,110 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final double INCREASING_RATE = 1.5;
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] internalArray;
+    private int size = 0;
+
+    public ArrayList() {
+        internalArray = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        size += 1;
+        if (size > internalArray.length) {
+            increaseCapacity();
+        }
+        internalArray[size - 1] = value;
```

Comment:
```suggestion
        if (size > internalArray.length) {
            increaseCapacity();
        }
        internalArray[size] = value;
        size++;
```

I think It looks better this way